test/gtest/ucp/test_ucp_mmap.cc

```diff
@@ -163,8 +163,12 @@ void test_ucp_mmap::test_rkey_management(entity *e, ucp_mem_h memh,
         }
     }
 
-    if (expect_rma_offload && is_dummy) {
-        EXPECT_NE(&ucp_rma_sw_proto, rkey->cache.rma_proto);
+    if (expect_rma_offload) {
+        if (is_dummy) {
+            EXPECT_EQ(&ucp_rma_sw_proto, rkey->cache.rma_proto);
+        } else {
+            EXPECT_EQ(&ucp_rma_basic_proto, rkey->cache.rma_proto);
+        }
```

Comment:
maybe just
```
if (expect_rma_offload && !is_dummy) {
    EXPECT_EQ(&ucp_rma_basic_proto, rkey->cache.rma_proto);
}
```

Reply:
maybe worth checking that in the case of dummy buffer we select SW proto?

Reply:
maybe this?
```
if (expect_rma_offload && !is_dummy) {
    EXPECT_EQ(&ucp_rma_basic_proto, rkey->cache.rma_proto);
} else {
  EXPECT_EQ(&ucp_rma_sw_proto, rkey->cache.rma_proto);
```

Reply:
unfortunately, it doesn't work for `shm_ib` test variant, since we couldn't expect whether we could use or not RMA offload. But if we could, it means whether IB (RC or DC) was selected or mmap method=allocation and SHM was selected.